include/zephyr/kernel.h

```diff
@@ -246,6 +246,18 @@ void k_thread_foreach_unlocked(
  */
 #define K_SSE_REGS (BIT(7))
 
+/**
+ * @brief HiFi registers are managed by context switch
+ *
+ * @details
+ * This option indicates that the thread uses the HiFi AudioEngine coprocessor
+ * registers. This instructs the kernel to save and restore the contents of
+ * these registers when scheduling the thread. No effect if
+ * @kconfig{CONFIG_XTENSA_HIFI_SHARING} is not enabled.
+ */
+#define K_HIFI_IDX 7
+#define K_HIFI_REGS (BIT(K_HIFI_IDX))
```

Comment:
Don't think it is necessary... as the code is saving/restoring as long as kconfig is enabled. There is no per thread settings.

Reply:
Nice catch. I was originally going to try for a per-thread setting but wound up not since there did not seem to be a clean way I recognised to get the pointer to the thread.

 Will remove.